### PR TITLE
Improve MySQL installation document.

### PR DESCRIPTION
**Pull request by berkerpeksag**

### CODE REVIEW

File: docs/topics/install-olympia/installation.rst [hunk 1]
```diff
@@ -181,14 +181,30 @@ git.
 Database
 --------
 
-The database is initialized automatically using the `make full_init` command
-you saw earlier. If you want to start from scratch and recreate the database,
-you can just run the `make initialize_db` command.
+First of all, you'll need to create a database named ``olympia`` and grant
+permissions to the user, based on your database settings in your
+``local_settings.py``::
```
Comment (magopian) [inline docs/topics/install-olympia/installation.rst]: This is correct, but I'd like a note specifying that it's optional. By default, the database used will be `olympia`, with the `root` user (without any password): https://github.com/mozilla/olympia/blob/master/lib/settings_base.py#L86

I'm not sure, but I think mysql installs by default with a passwordless root user, so you shouldn't have anything special to do, apart from creating the `olympia` database.

I would thus split this explanation in two parts:
1/ here's how to create the `olympia` database
2/ here's how to customize the user if you don't want to use the `root` user (with a more detailed explanation of how to specify the database url in the `local_settings.py` file)

**Comment by magopian** [comment]
This is excellent work @berkerpeksag! If you could apply the change above, I'll merge the result.

Thanks!

File: docs/topics/install-olympia/installation.rst (hunk 1 above)
Comment (berkerpeksag) [inline docs/topics/install-olympia/installation.rst]: Done. Thanks for the suggestions.

File: docs/topics/install-olympia/installation.rst [hunk 2]
```diff
@@ -177,18 +176,47 @@ It's awesome, you want it.
...
+your :ref:`local_settings.py<example-settings>` or set the environment variable
+``DATABASE_URL``::
+
+    export DATABASE_URL=mysql://<user>:<password>@<hostname>/<database>
+
+If you've changed the user and password information, you need to grant permissions
+to the new user::
+
+    $ mysql -u root -p
+    mysql> GRANT ALL ON olympia.* TO olympia@localhost IDENTIFIED BY '<YOUR_PASSWORD>';
```
Comment (magopian) [inline docs/topics/install-olympia/installation.rst]: I think you need to replace `olympia@localhost` by `<YOUR_USER>@localhost`, or explain that the example is for the user named `olympia`

File: docs/topics/install-olympia/installation.rst [hunk 3]
```diff
@@ -177,18 +176,47 @@ It's awesome, you want it.
...
+
+Finally, to run the test suite, you'll need to add an extra grant in MySQL for
+your database user::
+
+    $ mysql -u root -p
+    mysql> GRANT ALL ON test_olympia.* TO olympia@localhost IDENTIFIED BY '<YOUR_PASSWORD>';
```
Comment (magopian) [inline docs/topics/install-olympia/installation.rst]: Same as above

File: docs/topics/install-olympia/installation.rst [hunk 4]
```diff
@@ -177,18 +176,47 @@ It's awesome, you want it.
...
+
+.. warning::
+
+   Don't forget to change ``<YOUR_PASSWORD>`` to your actual database password.
+
+The database is initialized automatically using the ``make full_init`` command
+you saw earlier.
+
+If you want to start from scratch and recreate the database, you can just run
```
Comment (magopian) [inline docs/topics/install-olympia/installation.rst]: Maybe reword to something like `If, at some point, you want to start from...` so it's clear that it's not needed to do that after a `make full_init`

File: docs/topics/install-olympia/installation.rst [hunk 5]
```diff
@@ -198,10 +226,13 @@ Each incremental change we add to the database is done with a versioned SQL
 (and sometimes Python) file. To keep your local DB fresh and up to date, run
 migrations like this::
 
-    schematic migrations
+    $ schematic migrations/
 
-More info on schematic: https://github.com/mozilla/schematic
+If you want to start from scratch and recreate the database, you can just run
```
Comment (magopian) [inline docs/topics/install-olympia/installation.rst]: Maybe you don't need the previous identical sentence above, if you have it here?

**Comment by magopian** [comment]
Thanks @berkerpeksag we're getting there, just a few more minor nits ;)

File: docs/topics/install-olympia/installation.rst (hunk 2 above)
Comment (berkerpeksag) [inline docs/topics/install-olympia/installation.rst]: Done.

File: docs/topics/install-olympia/installation.rst (hunk 4 above)
Comment (berkerpeksag) [inline docs/topics/install-olympia/installation.rst]: Done.

File: docs/topics/install-olympia/installation.rst (hunk 5 above)
Comment (berkerpeksag) [inline docs/topics/install-olympia/installation.rst]: Done.

File: docs/topics/install-olympia/installation.rst (hunk 3 above)
Comment (berkerpeksag) [inline docs/topics/install-olympia/installation.rst]: Done.

**Comment by magopian** [comment]
This is perfect, thanks!
r+
